library/swift/src/Element.swift

```diff
@@ -0,0 +1,19 @@
+import Foundation
+
+private let kPattern = "^[A-Za-z_]+$"
+
+/// Element represents one dot-delimited component of a timeseries name.
+@objc
+public final class Element: NSObject, ExpressibleByStringLiteral {
+  private let value: String
+
+  public init(stringLiteral value: String) {
+    guard value.range(of: kPattern, options: .regularExpression) != nil else {
+      preconditionFailure("Element values must conform to the regex /^[A-Za-z_]+$/.")
+    }
+    self.value = value
+  }
+
+  // CustomStringConvertible
+  public override var description: String { return value }
```

Comment:
Why not make `value` internal? I think using `description` is fragile since it could break if someone deleted the explicit `description` override (yet would still compile)

Reply:
Okay, sure.